docs/src/rfc/text/0000-esrt.md

```diff
@@ -0,0 +1,383 @@
+manipulation. Type-safe enums represent firmware types and status codes, preventing invalid values
+at compile time.
+
+## Guide-Level Explanation
+
+Platform components receive the ESRT service as a parameter and use it to manage firmware resource
+metadata throughout the boot process. The ESRT table is automatically published on the Ready To Boot
+event - platforms do not need to call `publish_table()` explicitly.
+
+```rust
+use patina::component::service::Service;
+
+fn platform_init(esrt: Service<dyn EsrtRecords>) -> Result<()> {
+    // Register system firmware resource
+    let system_fw = SystemResourceEntry {
+        fw_class: SYSTEM_FIRMWARE_GUID,
+        fw_type: FirmwareType::SystemFirmware,
+        fw_version: 0x00010002,
+        lowest_supported_fw_version: 0x00010000,
+        capsule_flags: CAPSULE_FLAGS_PERSIST_ACROSS_RESET,
+        last_attempt_version: 0,
+        last_attempt_status: LastAttemptStatus::Success,
+    };
+    esrt.register_entry(&system_fw)?;
+
+    // After device enumeration, sync FMP resources
+    esrt.sync_fmp()?;
+
+    // Update entry after firmware update attempt
+    let mut entry = esrt.get_entry(&SYSTEM_FIRMWARE_GUID)?;
+    entry.last_attempt_version = new_version;
+    entry.last_attempt_status = LastAttemptStatus::Success;
+    esrt.update_entry(&entry)?;
+
+    // Lock before OS handoff
+    esrt.lock_repository()?;
+    
+    // Table will be automatically published on Ready To Boot event
+
+    Ok(())
+}
+```
+
+## Appendix A: Public API Summary
+
+This appendix lists only the externally visible, intentionally supported surface. Internal helper types
+(variable storage, repository structures, FFI glue) are intentionally excluded.
+
+### Configuration
+
+```rust
+pub struct EsrtConfiguration {
+    pub max_fmp_entries: u32,     // Maximum FMP entries, default platform-specific
+    pub max_non_fmp_entries: u32, // Maximum Non-FMP entries, default platform-specific
+}
+```
+
+### Core Types
+
+```rust
+/// Firmware resource entry
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SystemResourceEntry {
+    pub fw_class: Guid,
+    pub fw_type: FirmwareType,
+    pub fw_version: u32,
+    pub lowest_supported_fw_version: u32,
+    pub capsule_flags: u32,
+    pub last_attempt_version: u32,
+    pub last_attempt_status: LastAttemptStatus,
+}
+
+/// Firmware type classification
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FirmwareType {
+    Unknown,
+    SystemFirmware,
+    DeviceFirmware,
+    UefiDriver,
+}
+
+/// Last update attempt status
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LastAttemptStatus {
+    Success,
+    ErrorUnsuccessful,
+    ErrorInsufficientResources,
+    ErrorIncorrectVersion,
+    ErrorInvalidFormat,
+    ErrorAuthError,
+    ErrorPowerEventAc,
+    ErrorPowerEventBattery,
+    ErrorUnsatisfiedDependencies,
+    // Additional status codes per UEFI spec

```

Comment:
There are also Oem-defined proprietary LastAttemptStatus codes, so I suspect you're going to need an entry in this enum that is something like `VendorDefined(u32)`